app/domain/authentication/authn_k8s/authentication_request.rb

```diff
@@ -0,0 +1,32 @@
+module Authentication
+  module AuthnK8s
+
+    # This class validates the k8s authentication request against the resource
+    # restrictions configured in the policy for the requesting host.
+    # It uses the 'validate_k8s_resource' command-class to validate a k8s
+    # resource against the k8s API.
+    # This command-class receives the resource's type and name as input.
+    class AuthenticationRequest
+      def initialize(namespace:, validate_k8s_resource:)
+        @namespace = namespace
+        @validate_k8s_resource = validate_k8s_resource
```

Comment:
`validate_k8s_resource_class`?

Reply:
1. this is not a class, this is an instance of the command class.
2. we don't add `_class` suffix to command classes.

Reply:
ok, i thought it was a class